File: react/src/services/jobService.js
import axios from "axios";

const API_URL = 'http://localhost:8080/job';

export const createJob = async (postJobData) => {
    const formData = new FormData();
    formData.append("title" , postJobData.title);
    formData.append("description" , postJobData.description);
    formData.append("eligibility" , postJobData.eligibility);
    formData.append("skills" , postJobData.skills);
    formData.append("about" , postJobData.about);
    formData.append("numberOfOpenings" , postJobData.numberOfOpenings);
    formData.append("location" , postJobData.location);
    formData.append("ctc" , postJobData.ctc);
    formData.append("modeOfJob" , postJobData.modeOfJob);
    formData.append("applyBy" , postJobData.applyBy);
    formData.append("experience" , postJobData.experience);
    formData.append("startDate" , postJobData.startDate);
    formData.append("company" , postJobData.company);
    formData.append("domain" , postJobData.domain);
    formData.append("imageFile" , postJobData.imageFile);

    try{
        const response = await axios.post(
            API_URL+"/createJob" , 
            formData,{
            headers: {
                "Content-Type": "multipart/form-data",
              },
              responseType: "text",
    });
        if(response.status == 201){
            return response.data;
        }else{
            return "ERROR CREATING";
        }
    }
    catch(error){
        console.error('Error posting Job' , error);
        console.error('Error status' , error.response?.status);
        throw error;
    }
    
};

export const getAllJobs = async() =>{
    try{
    const response = await axios.get(
        API_URL+"/getAllJobs" 
    );
    return response.data;
    }catch(error){
        throw error;
    }
};

File: backend/src/main/java/com/internprep/InternPrepApplication/Service/JobService.java
package com.internprep.InternPrepApplication.Service;

import java.io.IOException;
import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.internprep.InternPrepApplication.Dto.JobEntityDto;
import com.internprep.InternPrepApplication.Entity.JobEntity;
import com.internprep.InternPrepApplication.Repository.JobRepository;

@Service
public class JobService {

    @Autowired
    private CloudinaryService cloudinaryService;
    @Autowired
    private JobRepository jobRepository;

    public String createJob(JobEntityDto jobEntityDto) throws IOException{
        String imageUrl = cloudinaryService.uploadFile(jobEntityDto.getImageFile());
        JobEntity jobEntity = JobEntity.builder()
        .company(jobEntityDto.getCompany())
        .about(jobEntityDto.getAbout())
        .applyBy(jobEntityDto.getApplyBy())
        .eligibility(jobEntityDto.getEligibility())
        .modeOfJob(jobEntityDto.getModeOfJob())
        .skills(jobEntityDto.getSkills())
        .numberOfOpenings(jobEntityDto.getNumberOfOpenings())
        .description(jobEntityDto.getDescription())
        .experience(jobEntityDto.getExperience())
        .imageUrl(imageUrl)
        .domain(jobEntityDto.getDomain())
        .location(jobEntityDto.getLocation())
        .startDate(jobEntityDto.getStartDate())
        .ctc(jobEntityDto.getCtc())
        .title(jobEntityDto.getTitle())
        .build();

        JobEntity createdJob = jobRepository.save(jobEntity);
        if(createdJob != null && createdJob.getId() != null){
            return "Job Created Successfully";
        }else{
            return "Failed to create Job";
        }
    }

    public List<JobEntity> getAllJobs(){
        List<JobEntity> allJobs = jobRepository.findAll();
        if(allJobs.size() > 0){
            return allJobs;
        }else{
            return null;
    }
    }

}
